pkg/defaults/defaults.go

```diff
@@ -803,3 +812,34 @@ func buildRootImageStreamFromRepository(readFile readFile) (*api.ImageStreamTagR
 	}
 	return &config.BuildRootImage, nil
 }
+
+func ensureImageStreamTag(ctx context.Context, client ctrlruntimeclient.Client, isTagRef *api.ImageStreamTagReference, second time.Duration) {
+	istImport := &testimagestreamtagimportv1.TestImageStreamTagImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      isTagRef.Name + "-" + isTagRef.Tag,
+			Namespace: "ci",
+		},
+		Spec: testimagestreamtagimportv1.TestImageStreamTagImportSpec{
+			Namespace: isTagRef.Namespace,
+			Name:      isTagRef.Name + ":" + isTagRef.Tag,
+		},
+	}
+	istImport.SetImageStreamLabels()
+
+	// Conflicts are expected
+	if err := client.Create(ctx, istImport); err != nil && !kapierrors.IsConflict(err) {
+		logrus.WithError(err).Warnf("Failed to create imagestreamtagimport for root %s", istImport.Name)
+	}
+
+	if err := wait.PollImmediate(5*second, 30*second, func() (bool, error) {
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKeyFromObject(istImport), &imagev1.ImageStreamTag{}); err != nil {
```

Comment:
Why is this not a watch?

Reply:
Because the imagestreamtag api does not support the watch verb

Reply:
```
Error from server (MethodNotAllowed): the server does not allow this method on the requested resource (get imagestreamtags.image.openshift.io)
```

Reply:
Oh, right ...